ibuffer-project.el

```diff
@@ -63,18 +63,34 @@
 (require 'ibuffer)
 (require 'ibuf-ext)
 
+(defvar ibuffer-project-group-cache (make-hash-table)
+  "Cache for mappings from a directory to its group data.")
+
+(defun ibuffer-project-purge-group-cache ()
+  "Purge the group cache."
+  (setq ibuffer-project-group-cache (make-hash-table)))
+
 (defun ibuffer-project-root (buf)
   "Return a cons cell (project-root . root-type) for BUF."
   (unless (string-match-p "^ " (buffer-name buf))
-    (with-current-buffer buf
-      (let ((dir (cdr (project-current))))
-        (cond
-         (dir (cons dir 'project))
-         (default-directory (cons (abbreviate-file-name default-directory) 'directory)))))))
+    (let ((dir (buffer-local-value 'default-directory buf)))
+      (when dir
+        (or (gethash dir ibuffer-project-group-cache)
+            (let* ((root (cdr (project-current nil dir)))
+                   (result (cond
+                            (root (cons root 'project))
+                            (dir (cons (abbreviate-file-name dir) 'directory)))))
+              (when result
```

Comment:
I think this check is not necessary. `dir` is checked at line 77 so `result` can't be `nil` here.

Reply:
FIxed.